cdragontoolbox/tftdata.py

```diff
@@ -180,15 +255,20 @@ def parse_traits(self, map22):
 
             effects = []
             for trait_set in trait.getv("mTraitSets"):
-                variables = []
+                variables = {}
+                increment = 0
                 for effect in trait_set.getv("effectAmounts", []):
-                    variables.append({"name": effect.getv("name"), "value": effect.getv("value")})
+                    name = effect.getv("name").s if effect.__contains__("name") else "unknown" + str(increment)
```

Comment:
- use `in` instead of `__contains__`
- Use f-string: `f"unknown{increment}"`

Reply:
this should use the hash path instead of unknown, i just couldnt be bothered to find what variable it was hidden in

Reply:
I don't see the link with my comments (especially the one about `__contains__`).